Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -4,12 +4,35 @@ import MapboxCoreNavigation
 import MapboxMaps
 import MapboxDirections
 
-// :nodoc:
+/**
+ `PreviewViewController` is a user interface for the preview mode experience. It allows to present
+ banners that contain information that is important for the user before actually starting turn-by-turn
+ navigation session (e.g. final destination presentation, main and alternative routes preview etc).
+ By default Mapbox Navigation SDK provides such banners:
+ - `DestinationPreviewViewController` - banner that is shown at the bottom of the screen and allows
+ to show information about the final destination, preview available routes and start active navigation
+ session
+ - `RoutesPreviewViewController` - banner that is shown at the bottom of the screen and allows to
+ preview information about the current `Route` (expected travel time, distance and expected time of arrival)
+ - `BannerDismissalViewController` - banner that is shown at the top of the screen and allows to
+ dismiss already presented banner
+ 
+ Internally `PreviewViewController` relies on two components:
+ - `NavigationMapView` - wraps `MapView` and provides convenience functions for adding and removing
+ route lines, route duration annotations, shows user location indicator etc
+ - `NavigationView` - wraps `NavigationMapView` and provides the ability to show drop-in related UI
+ components like `SpeedLimitView`, `WayNameView`, top and bottom `BannerContainerView`s etc
+ 
+ `PreviewViewController` works as an initial step before switching to the active navigation.
+ Use `NavigationViewController` for turn-by-turn navigation experience.
+ */
 public class PreviewViewController: UIViewController, BannerPresentation {
```

Comment:
I wonder should `BannerPresentation` be marked public and documented or should the conformance to `BannerPresentation` moved to internal extension?

Reply:
Intention is to keep `BannerPresentation` internal and its default implementation until we move other components to it (most important one is `NavigationViewController`), after that (and possibly changes to default implementation) we can make it public. Thanks for suggestion, I've moved conformance to `BannerPresentation` out of sight to the internal section.